floris/tools/energy_ratio.py

```diff
@@ -463,35 +472,63 @@ def calculate_balanced_energy_ratio(
         percentiles = _get_confidence_bounds(confidence)
 
         # Compute the central over from the bootstrap runs
-        ratio_array_base[i] = np.mean(ratio_base_bs)
-        ratio_array_con[i] = np.mean(ratio_con_bs)
-        diff_array[i] = np.mean(diff_bs)
-        p_change_array[i] = np.mean(p_change_bs)
-
-        lower_ratio_array_base[i], upper_ratio_array_base[
-            i] = _calculate_lower_and_upper_bound(
-                ratio_base_bs,
-                percentiles,
-                central_estimate=ratio_array_base[i],
-                method='simple_percentile')
-        lower_ratio_array_con[i], upper_ratio_array_con[
-            i] = _calculate_lower_and_upper_bound(
-                ratio_con_bs,
-                percentiles,
-                central_estimate=ratio_array_con[i],
-                method='simple_percentile')
-        lower_diff_array[i], upper_diff_array[
-            i] = _calculate_lower_and_upper_bound(
-                diff_bs,
-                percentiles,
-                central_estimate=diff_array[i],
-                method='simple_percentile')
-        lower_p_change_array[i], upper_p_change_array[
-            i] = _calculate_lower_and_upper_bound(
-                p_change_bs,
-                percentiles,
-                central_estimate=p_change_array[i],
-                method='simple_percentile')
+        # print('wd',wind_direction_bin)
+        # print(ratio_base_bs)
+        # print(np.mean(ratio_base_bs))
+
+        # Trim out the nans
+        # print('size before',len(ratio_base_bs))
```

Comment:
Do we know how often NANs occur? I think we can push to remove this kind of code by addressing the root cause.

Reply:
Fairly certain all nans I've seen so far are fully-empty bins, which the new code handles better.  Still, it's all a bit experimental would it be ok to leave some of this for now?

Reply:
Yea, for sure, we can leave it for now, but something we should definitely work on handling differently.

Reply:
works for me!